src/main/java/thePackmaster/potions/SmithingOil.java

```diff
@@ -0,0 +1,74 @@
+package thePackmaster.potions;
+
+
+import basemod.abstracts.CustomPotion;
+import com.badlogic.gdx.graphics.Color;
+import com.evacipated.cardcrawl.mod.stslib.actions.common.SelectCardsInHandAction;
+import com.megacrit.cardcrawl.actions.common.UpgradeSpecificCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.helpers.PowerTip;
+import com.megacrit.cardcrawl.localization.PotionStrings;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.util.Wiz;
+
+import static thePackmaster.util.Wiz.atb;
+
+
+public class SmithingOil extends CustomPotion {
```

Comment:
This is an extremely powerful effect. I know I'd be very excited to see an upgrade drop from a fight, which is essentially what this is. But it's so powerful that I wonder if it's too much to include. Thoughts?

If we do keep this, since this potion has a permanent benefit, this will need a patch for `AbstractDungeon.returnRandomPotion`. We need the same check as for Fruit Juice (not allowing it when `limited` is true), so that this potion can't be randomly generated by Alchemize or any other effect like that.

Reply:
Yeah, the comparison to Fruit Juice is what I was going for here.  I'm open to other ideas.  The Smithing Hammer gave an 'in' for theming to Packmaster, and I wanted the Rare to be as generally useful as possible.